File: learning_platform_api_gateway/internal/dto/lessonDto/updateLesson.go
package lessonDto

import (
	"time"
)

type UpdateLesson struct {
	ID              int64       `json:"id"`
	BoardID         *int64      `json:"board_id"`
	MeetLink        *string     `json:"meet_link"`
	StartTime       time.Time   `json:"start_time"`
	Duration        int64       `json:"duration"`
	MediaItems      []MediaItem `json:"media_items"`
	UserIDs         []int64     `json:"user_ids"`
	DeletedUserIDs  []int64     `json:"deleted_user_ids"`
	DeletedMediaIDs []int64     `json:"deleted_media_ids"`
}

File: learning_platform_front/src/entities/lesson/types/updateLesson.ts
import type { CreateLessonMediaItemRequest } from "./createLesson"

export type UpdateLessonRequest = {
    id: number
    board_id?: number | null
    meet_link?: string | null
    start_time: string
    duration: number
    media_items?: CreateLessonMediaItemRequest[]
    user_ids?: number[]
    deleted_user_ids?: number[]
    deleted_media_ids?: number[]
}
